pkg/asset/installconfig/powervs/session.go

```diff
@@ -182,6 +185,67 @@ func (c *BxClient) ValidateAccountPermissions() error {
 	return nil
 }
 
+//ValidateDhcpService checks for existing dhcp instance for the provided PowerVS cloud instance
+func (c *BxClient) ValidateDhcpService(ctx context.Context, svcInsID string) error {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, helpers.PIGetTimeOut)
+	defer cancel()
+	res := make(chan int)
+
+	go func() error {
+		//create Power VS DHCP Client
+		dhcpClient := instance.NewIBMPIDhcpClient(ctxWithTimeout, c.PISession, svcInsID)
+		//Get all DHCP Services
+		dhcpServers, err := dhcpClient.GetAll()
+		if err != nil {
+			return err
+		}
+		res <- len(dhcpServers)
+		close(res)
+		return nil
+	}()
+	for {
+		select {
+		case noOfDhcpServices := <-res:
+			if noOfDhcpServices > 0 {
+				return fmt.Errorf("DHCP service already exists for provided cloud instance")
+			}
+			return nil
+		case <-ctxWithTimeout.Done():
+			return ctxWithTimeout.Err()
+		}
+	}
+}
+
+//ValidateCloudConnectionInPowerVSRegion counts cloud connection in PowerVS Region
+func (c *BxClient) ValidateCloudConnectionInPowerVSRegion(ctx context.Context, svcInsID string) error {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, helpers.PIGetTimeOut)
+	defer cancel()
+	res := make(chan int)
+
+	go func() error {
+		//check number of cloudconnection must be less than or equals to one
+		ccClient := instance.NewIBMPICloudConnectionClient(ctxWithTimeout, c.PISession, svcInsID)
+		getAllResp, err := ccClient.GetAll()
+		if err != nil {
+			return err
+		}
+		res <- len(getAllResp.CloudConnections)
+		close(res)
+		return nil
+	}()
+	for {
+		select {
+		case noOfCloudConnections := <-res:
+			if noOfCloudConnections >= 1 {
+				return fmt.Errorf("only one cloudconnection with dhcpservice is allowed per Zone")
```

Comment:
1) I don't see where this checks that there is a DHCP service associated with this Cloud connection. The existence of another Cloud connection on its own isn't a problem.
2) Cloud connection is two words. All of our documentation capitalizes the first word but not the second: https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-cloud-connections
3) DHCP service is two words. DHCP should always be capitalized as it is an acronym. 
4) Zone doesn't need to be capitalized. 
5) The error isn't exactly clear to an end user. Change this to something like, "only one Cloud connection can be attached to any DHCP network per account per zone."